src/libp2p/Libp2p.Core/Stream.cs

```diff
@@ -0,0 +1,106 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: MIT
+
+using Microsoft.Extensions.Logging;
+using Nethermind.Libp2p.Core;
+using System.Buffers;
+
+public class ChannelStream : Stream
+{
+    private readonly IChannel chan;
+    private readonly ILogger<ChannelStream> logger;
+    private bool _disposed = false;
+    private bool _canRead = true;
+    private bool _canWrite = true;
+
+    // Constructor
+    public ChannelStream(IChannel _chan, ILogger<ChannelStream> _logger = null)
```

Comment:
You can omit logger, it's not used. Could you also name `_chan` argument as just `chan` and use a [primary constructor](https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/tutorials/primary-constructors).

Reply:
done